README.md

```diff
@@ -1,51 +1,199 @@
+Showing incoming call notification/screen using iOS CallKit and Android Custom UI for Flutter
 
-## Getting Started
+## Native setup
 
-This project is a starting point for a Flutter
-[plug-in package](https://flutter.dev/developing-packages/),
-a specialized package that includes platform-specific implementation code for
-Android and/or iOS.
+flutter_callkeep requires the following permissions.
 
-For help getting started with Flutter, view our 
-[online documentation](https://flutter.dev/docs), which offers tutorials, 
-samples, guidance on mobile development, and a full API reference.
+### Android
 
-## Device Permission
+No extra setup is needed
 
-flutter_callkeep requires the following permissions.
 
-### Android
+### iOS
+in `Info.plist`
 
-If you want to use the function `displayIncomingCall`, please add the following permissions and service to the `AndroidManifest.xml`.
+```
+<key>UIBackgroundModes</key>
+<array>
+    <string>processing</string>
+    <string>remote-notification</string>
+    <string>voip</string>
+</array>
+```
 
-```xml
-..
-<uses-permission android:name="android.permission.BIND_TELECOM_CONNECTION_SERVICE" />
-<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
-<uses-permission android:name="android.permission.READ_PHONE_STATE" />
-<uses-permission android:name="android.permission.CALL_PHONE" />
-<uses-permission android:name="android.permission.RECORD_AUDIO" />
-..
+Then you need to update `AppDelegate.swift` to follow the example for handling PushKit as push handling must be done through native iOS code due to [iOS 13 PushKit VoIP restrictions](https://developer.apple.com/documentation/pushkit/pkpushregistrydelegate/2875784-pushregistry).
+
+```swift
+import UIKit
+import PushKit
+import Flutter
+import flutter_callkeep
+
+@UIApplicationMain
+@objc class AppDelegate: FlutterAppDelegate, PKPushRegistryDelegate {
+    override func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+    ) -> Bool {
+        GeneratedPluginRegistrant.register(with: self)
+        
+        //Setup VOIP
+        let mainQueue = DispatchQueue.main
+        let voipRegistry: PKPushRegistry = PKPushRegistry(queue: mainQueue)
+        voipRegistry.delegate = self
+        voipRegistry.desiredPushTypes = [PKPushType.voIP]
+        
+        return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+    }
+    
+    // Call back from Recent history
+    override func application(_ application: UIApplication,
+                              continue userActivity: NSUserActivity,
+                              restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
+        
+        guard let handleObj = userActivity.handle else {
+            return false
+        }
+        
+        guard let isVideo = userActivity.isVideo else {
+            return false
+        }
```

Comment:
Why only start call if it's video?

Reply:
This is only the README, but I agree this looks bad, updating it to reflect a real-world scenario.